roles/elasticsearch/tasks/main.yml

```diff
@@ -38,5 +38,13 @@
 
 - name: register elasticsearch firewall ports
   set_fact:
-    firewall_ports: >
-      {{ firewall_ports + [elasticsearch_port] }}
+    firewall_data: >
+      {{ firewall_data }} + [
+        {
+          'source': "{{ item }}",
```

Comment:
Shouldn't this be "destination"? The `elasticsearch_interface_ipv4` address is not where connections are coming *from*; it's where clients are connecting *to*.